internal/state/rollback.go

```diff
@@ -37,7 +37,7 @@ func Rollback(bs BlockStore, ss Store) (int64, []byte, error) {
 
 	// state store height is equal to blockstore height. We're good to proceed with rolling back state
 	rollbackHeight := invalidState.LastBlockHeight
-	rollbackBlock := bs.LoadBlockMeta(rollbackHeight)
+	rollbackBlock := bs.LoadBlockMeta(rollbackHeight - 1)
```

Comment:
ah, this should really be
```
rollbackHeight := invalidState.LastBlockHeight - 1
rollbackBlock := bs.LoadBlockMeta(rollbackHeight)
```

and we'll need to update the calls below that are using the `rollbackHeight`

Reply:
Line 45 `previousValidatorSet, err := ss.LoadValidators(rollbackHeight - 1)` is using the pred block height. Should `previousParams, err := ss.LoadConsensusParams(rollbackHeight)` (line 50) be using the same or is that right?